mongomock/aggregate.py

```diff
@@ -634,6 +635,62 @@ def _handle_date_operator(self, operator, values):
                     ' in Mongomock.'
                 )
             return out_value['date'].strftime(out_value['format'])
+        if operator == '$dateFromParts':
+            if not isinstance(values, dict):
+                raise OperationFailure(
+                    '$dateFromParts operator must correspond a dict '
+                    'that has "year" or "isoWeekYear" field.'
+                )
+            if (
+                not isinstance(values, dict)
+                or len(set(values.keys()).intersection({'year', 'isoWeekYear'})) != 1
+            ):
+                raise OperationFailure(
+                    f'{operator} operator must correspond a dict '
+                    'that has "year" or "isoWeekYear" field.'
+                )
+            if 'isoWeekYear' in values.keys():
+                raise NotImplementedError(
+                    'Although isoWeekYear is a valid field for the '
+                    f'{operator} operator, it is currently not implemented '
+                    ' in Mongomock.'
```

Comment:
nit - please drop the blank before `in` as you already have one at the end of the preceding line.